### ISSUE TITLE
Tedit edited world crashes in-game

### ISSUE THREAD

**Issue by DZDXY** [TEdit/Terraria-Map-Editor#1083]
## Before you submit an issue, please read
 * If you get an unhandled exception, please _**check your log before posting**_.
 * If your log includes _**DXGI_ERROR_DEVICE_REMOVED/DeviceRemoved**_
   * This is a known issue, do not create a new issue, please add your comments to #751
   * People have had the most luck in Windows 7 with updating to DirectX 11.1
     This is found in windows update in the optional updates section or by
     following directions here 
   * https://support.microsoft.com/en-us/kb/179113
 * Please fill out the questions below
 * Instead of copy/pasting text you can attach the entire log file.
 * If your world file is less than 25MB, you can attach it as well.
   
## What were you doing when the issue occured?

The world I have edited with Tedit always crashes shortly after loading up in Terraria itself. To be more specific, I can successfully load my edited world in Terraria but I can only play for a matter of seconds before Terraria crashes and closes completely. I can load the world fine in Tedit so I'm guessing it's not a corrupted file, but I don't know how to be sure. I've verified integrity via Steam but that didn't seem to do anything. I am posting this here because all of my other worlds (which have not been edited using Tedit) are working as one would expect so I can only assume something I did in Tedit has caused this problem/

## Can you reproduce the issue?

I would if I knew exactly what the issue is. All I can really say is that I had added a lot of wiring and only after that did my world start crashing in-game. So I went and erased most of it to see what happened and it did seem to lengthen my play time from a few seconds to a couple minutes but the crashing persisted.

## What version of Terraria are you using?

1.3.5.3

## Post relevant contents of log or attach log file _**(very important)**_:

This is not a problem with Tedit as an application. I have the crash log for Terraria enabled but for some reason when this specific world crashes, nothing is added to the log.

**Comment by BinaryConstruct** [comment]
Can you attach a world file zip?

**Comment by DZDXY** [comment]
[yoyoyomama.zip](https://github.com/TEdit/Terraria-Map-Editor/files/2007301/yoyoyomama.zip)

**Comment by DZDXY** [comment]
I finally had a Terraria crash that resulted in a recorded log:

Error Logging Enabled.
================
5/18/2018 4:34:34 PM: First-Chance Exception
Culture: en-US
Exception: System.IO.IOException: The handle is invalid.

   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
================

Resolution changed to: 1920x1080.
================
5/18/2018 4:34:52 PM: First-Chance Exception
Culture: en-US
Exception: System.DivideByZeroException: Attempted to divide by zero.
   at Terraria.Main.DrawSurfaceBG()
================

WaveMaskData texture recreated. 1x1
WaveMaskData texture recreated. 85x135
WaveMaskData texture recreated. 85x135
WaveMaskData texture recreated. 85x135
WaveMaskData texture recreated. 86x135
WaveMaskData texture recreated. 86x135
WaveMaskData texture recreated. 86x135
